software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+   -- since in this case there aren't any unsaved changes yet
+   put "opened" into sMain[sNowDoc][0]["saveStatus"]
+   
+   -- get the file path for this stack
+   --   put the effective filename of this stack into tPath
+   --   set the itemDelimiter to slash
+   --   -- if this is a standalone application running on a Mac, find the path to the .app file
+   --   if the platform = "MacOS" and the environment = "standalone application" then
+   --      repeat until last item of tPath contains ".app"
+   --         delete last item of tPath
+   --      end repeat
+   --   end if
+   --   -- if not on a Mac, remove the last item to get the path to the containing folder
+   --   --   delete last item of tPath
+   --   put "data" into item -1 of tPath -- use the data folder
+   --   --Ask the user to choose a file
+   --   answer file "Select a file to open" with tPath
+   --   if the result is not "Cancel" then
+   --      if it ends with ".how" or ".txt" then
+   --         put it into tChosenFile
+   --         --         userOpen tChosenFile
+   --         put URL ("file:" & tChosenFile) into tJason
+   --         put empty into sMain
+   --         put arrayFromJson(tJason) into sMain[1]
+   --         put 1 into sNowDoc
+   --         -- filepath
+   --         put tChosenFile into sMain[sNowDoc][0]["filepath"] -- absolute path to *.how file
+   --         put item -1 of tChosenFile into tForShow
+   --         replace ".how" with "" in tForShow
+   --         put tForShow into field "filenameLabel" stack "HowstrBeta"
+   --         refreshDraw
+   --         refreshDisplay
+   --      else
+   --         break
+   --      end if
+   --   end if
+   -- this will overwrite the "unsaved" that the build handler flags
+   -- since in this case there aren't any unsaved changes yet
+   --   put "opened" into sMain[sNowDoc][0]["saveStatus"]
+end loadButtMouseUp
+
+on saveAsButtMouseUp
+   -- get the file path for this stack
+   put the effective filename of this stack into tPath
+   set the itemDelimiter to slash
+   -- if this is a standalone application running on a Mac, find the path to the .app file
+   if the platform = "MacOS" and the environment = "standalone application" then
+      repeat until last item of tPath contains ".app"
+         delete last item of tPath
+      end repeat
+   end if
+   -- if not on a Mac, remove the last item to get the path to the containing folder
+   --   delete last item of tPath
+   
+   -- shouldn't put project folders into /data, it's inside the github dev repository
+   -- documents folder makes sense as a default, but allow for anywhere
+   put specialFolderPath("documents") into tDocsPath
+   put sMainShort[sNowDoc][2]["title"] into tCaseName
+   if tCaseName is empty then put "[blank]" into tCaseName
+   ask file "New project name (the case title is: " & tCaseName& ")"  with tDocsPath
+   if the result is not "cancel" then
+      put it into tReq["newPath"]
```

Comment:
@matthewmaier You may want to ensure the user included an extension. Something like this:

````
command EnsureExtension @pFilename, pExtension
    set the itemDelimiter to "."
    if item -1 of pFilename is not pExtension then
         -- handle the user entering in . but no extension
         if char -1 of pFilename is not "." then
             put "." after pFilename
         end if
         put "how" after pFilename
    end if
end EnsureExtension
````